sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JsonInferSchema.scala

```diff
@@ -99,6 +100,14 @@ private[sql] class JsonInferSchema(options: JSONOptions) extends Serializable {
             val wrappedCharException = new CharConversionException(msg)
             wrappedCharException.initCause(e)
             handleJsonErrorsByParseMode(parseMode, columnNameOfCorruptRecord, wrappedCharException)
+          case e: FileNotFoundException if options.ignoreMissingFiles =>
+            logWarning(s"Skipped missing file", e)
+            Some(StructType(Nil))
+          // Throw FileNotFoundException even if `ignoreCorruptFiles` is true
+          case e: FileNotFoundException if !options.ignoreMissingFiles => throw e
+          case e: IOException if options.ignoreCorruptFiles =>
```

Comment:
`RuntimeException` already be catched in https://github.com/apache/spark/pull/42979/files/942b7c38a277a1b38b85a64ad940b56528ae8a03#diff-774d08eb04cd18039c576c7e23609430476d3dd2668535f0432f04b65b8ab234R92.

Reply:
It never reach here if we catch one `RuntimeException`. Because it will be catched by https://github.com/apache/spark/pull/42979/files/942b7c38a277a1b38b85a64ad940b56528ae8a03#diff-774d08eb04cd18039c576c7e23609430476d3dd2668535f0432f04b65b8ab234R92. So it would be useless code.

Reply:
oh, I got it. But I'm not sure it would change behavior or not. Let me change it and see CI would report error or not.